### PR TITLE
Feature/ws2 2949 locking add graphql support

### PR DESCRIPTION
**Pull request by JulissaDantes**
We have to wait for the Ceramic side to be ready, but we could start reviewing this side in the meantime. Here we add the new directive to recognize immutable fields and make sure to support parsing it.

### CODE REVIEW

**Review by PaulLeCam — COMMENTED** [review]
Looks great for the devtools part, but could you also update the runtime schema so that immutable fields are not present in the update inputs please?
The method generating these inputs is here: https://github.com/ceramicstudio/js-composedb/blob/main/packages/runtime/src/schema.ts#L866
I think that will also require some changes to the runtime definition to indicate what fields are immutable, here are the shared type definitions: https://github.com/ceramicstudio/js-composedb/blob/main/packages/types/src/index.ts

Let me know if you need more details or you want to sync on these changes some time!

File: packages/devtools/src/schema/parser.ts [hunk 1]
```diff
@@ -482,6 +488,7 @@ export class SchemaParser {
     directives: Array<DirectiveAnnotation>,
   ): DefinitionWithReferences<ItemDefinition> {
     const required = isNonNullType(type)
+    const immutable = directives.some((item) => item.name === 'immutable')
```
Comment (PaulLeCam) [inline packages/devtools/src/schema/parser.ts:491]: I think we probably need an additional check somewhere that the directive is only used on models (having the `@createModel` directive present), as we don't support it on nested objects.

Comment (JulissaDantes) [reply packages/devtools/src/schema/parser.ts:508]: Please check my "feat:check if nested object" commit, I take the object directives to check for the `createModel` value.

File: packages/devtools/test/schema.test.ts [hunk 2]
```diff
@@ -786,6 +786,90 @@ describe('schema parsing and compilation', () => {
...
+      createAbstractCompositeDefinition(`
+      interface TestInterface @loadModel(id: "interface ID") {
+        id: ID! @immutable
+      }
+
+      type TestModel @createModel(description: "Test model") {
+        text: String! @string(maxLength: 10)
+      }
+      `)
+    }).toThrow(`Unsupported immutable directive for id on nested object TestInterface`)
```
Comment (PaulLeCam) [inline packages/devtools/test/schema.test.ts:870]: The error message is misleading here, as `TestInterface` is not a nested object.
Also that makes me wonder: has the interface implementation validation logic been updated to make sure the models implementing an interface with immutable fields also set these fields as immutable please?

**Review by PaulLeCam — CHANGES_REQUESTED** [review]
LGTM other than the inherited immutable fields from the interfaces that could result in duplicate keys

**Pull request merged**